Documentation/gettingstarted/cni-chaining-aws-cni.rst

```diff
@@ -0,0 +1,131 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    http://docs.cilium.io
+
+*******
+AWS-CNI
+*******
+
+This guide explains how to set up Cilium in combination with aws-cni. In this
+hybrid mode, the aws-cni plugin is responsible for setting up the virtual
+network devices as well as address allocation (IPAM) via ENI. After the initial
+networking is setup, the Cilium CNI plugin is called attach BPF programs to the
+network devices set up by aws-cni to enforce network policies, perform
+load-balancing, and encryption.
+
+.. image:: aws-cni-architecture.png
+
+
+Setup Cluster on AWS
+====================
+
+Follow the instructions in the :ref:`k8s_install_eks` guide to set up an EKS
+cluster or use any other method of your preference to set up a Kubernetes
+cluster.
+
+Ensure that the `aws-vpc-cni-k8s <https://github.com/aws/amazon-vpc-cni-k8s>`_
+plugin is installed. If you have set up an EKS cluster, this is automatically
+done.
+
+Prepare Cilium to use AWS-CNI chaining
+======================================
+
+Download the Cilium deployment yaml:
+
+.. tabs::
+  .. group-tab:: K8s 1.14
+
+    .. parsed-literal::
+
+      curl -sLO \ |SCM_WEB|\/examples/kubernetes/1.14/cilium.yaml
+
+  .. group-tab:: K8s 1.13
+
+    .. parsed-literal::
+
+      curl -sLO \ |SCM_WEB|\/examples/kubernetes/1.13/cilium.yaml
+
+  .. group-tab:: K8s 1.12
+
+    .. parsed-literal::
+
+      curl -sLO \ |SCM_WEB|\/examples/kubernetes/1.12/cilium.yaml
+
+  .. group-tab:: K8s 1.11
+
+    .. parsed-literal::
+
+      curl -sLO \ |SCM_WEB|\/examples/kubernetes/1.11/cilium.yaml
+
+  .. group-tab:: K8s 1.10
+
+    .. parsed-literal::
+
+      curl -sLO \ |SCM_WEB|\/examples/kubernetes/1.10/cilium.yaml
+
+Edit ``cilium.yaml and add the following configuration to the ConfigMap:
+
+.. code:: bash
+
+      cni-chaining-mode: aws-cni
+      masquerade: "false"
+      tunnel: disabled
+
+This will enable chaining with the aws-cni plugin. It will also disable
+tunneling. Tunneling is not required as ENI IP addresses can be directly routed
+in your VPC. You can also disable masquerading for the same reason.
+
+Validate your Security Groups
+=============================
+
+Validate your AWS security groups rules and ensure that ENI IP addresses as
+allocated and used by the aws-cni plugin are allowed. See the documentation of
+the `aws-vpc-cni-k8s <https://github.com/aws/amazon-vpc-cni-k8s>`_ plugin for
+more details.
+
+Deploy Cilium
+=============
+
+.. code:: bash
+
+       kubectl apply -f cilium.yaml
+
+As Cilium is deployed as a DaemonSet, it will write a new CNI configuration
+``05-cilium.conflist`` which will take precedence over the standard
+``10-aws.conflist``. Any new pod scheduled, will use the chaining configuration
+which will not also invoke Cilium.
```

Comment:
maybe we should recommend deleting the `10-aws.conflist` to avoid the container runtime from using both?

Reply:
That's likely easier said than done, it will get re-created on various occasions. I've added a note to the docs though.